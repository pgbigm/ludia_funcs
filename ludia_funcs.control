# ludia_funcs extension
comment = 'ludia functions'
default_version = '1.0'
module_pathname = '$libdir/ludia_funcs'
relocatable = true
