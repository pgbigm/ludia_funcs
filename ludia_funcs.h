/*-------------------------------------------------------------------------
 *
 * Copyright (c) 2006-2015, NTT DATA Corporation
 * All rights reserved.
 *
 * Changelog:
 *   2013/01/09
 *   Update Ludia functions so that they are available with PostgreSQL9.1.
 *   Author: NTT DATA Corporation
 *
 *-------------------------------------------------------------------------
 */
#ifndef __LUDIA_FUNCS_H__
#define __LUDIA_FUNCS_H__

#ifdef TEXTPORTER
int	ExecTextPorter(unsigned char *Appfile, unsigned char *Txtfile,
				   unsigned char *GroupName, unsigned char *DefLangName,
				   int bBigEndian, unsigned int Option,
				   unsigned int Option1, long Size, unsigned short Csv_c);
#endif	/* TEXTPORTER */

#endif	/* __LUDIA_FUNCS_H */
