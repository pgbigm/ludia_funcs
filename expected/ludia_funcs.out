\pset null '(null)'
-- Load ludia_funcs module
LOAD '$libdir/ludia_funcs';
CREATE EXTENSION ludia_funcs;
SET standard_conforming_strings TO off;
SET escape_string_warning TO off;
-- pgs2snippet1() checks
SET ludia_funcs.escape_snippet_keyword TO on;
SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', '\\');
 pgs2snippet1 
--------------
 @\@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"￥"', '\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"￥"', '￥');
 pgs2snippet1 
--------------
 @￥@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', '￥');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', '"');
 pgs2snippet1 
--------------
 @"@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"”"', '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"”"', '”');
 pgs2snippet1 
--------------
 @”@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', '”');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', '\\\\');
 pgs2snippet1 
--------------
 @\@@\@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\\\"', '\\\\');
 pgs2snippet1 
--------------
 @\\@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\\\"', '\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', '""');
 pgs2snippet1 
--------------
 @"@@"@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""""', '""');
 pgs2snippet1 
--------------
 @""@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""""', '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', 'あ\\');
 pgs2snippet1 
--------------
 あ@\@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', '\\あ');
 pgs2snippet1 
--------------
 @\@あ
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ\\"', 'あ\\');
 pgs2snippet1 
--------------
 @あ\@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\あ"', '\\あ');
 pgs2snippet1 
--------------
 @\あ@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ\\"', '\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\あ"', '\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ\\"', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\あ"', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', 'あ"');
 pgs2snippet1 
--------------
 あ@"@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', '"あ');
 pgs2snippet1 
--------------
 @"@あ
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ""', 'あ"');
 pgs2snippet1 
--------------
 @あ"@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""あ"', '"あ');
 pgs2snippet1 
--------------
 @"あ@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ""', '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""あ"', '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ""', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""あ"', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う\\え"', 'う\\え');
 pgs2snippet1 
--------------
 @う\え@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う\\"', 'う\\え');
 pgs2snippet1 
--------------
 @う\@え
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\え"', 'う\\え');
 pgs2snippet1 
--------------
 う@\え@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\"', 'う\\え');
 pgs2snippet1 
--------------
 う@\@え
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う\\え"', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う\\"', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\え"', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う\\え"', 'う');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う"え"', 'う"え');
 pgs2snippet1 
--------------
 @う"え@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う""', 'う"え');
 pgs2snippet1 
--------------
 @う"@え
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""え"', 'う"え');
 pgs2snippet1 
--------------
 う@"え@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"""', 'う"え');
 pgs2snippet1 
--------------
 う@"@え
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う"え"', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う""', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""え"', 'うえ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"う"え"', 'う');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\\\"', 'あいう\\えお');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\\\"', 'あいう\\\\えお');
  pgs2snippet1  
----------------
 あいう@\\@えお
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\""', 'あいう"えお');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"\\""', 'あいう\\"えお');
  pgs2snippet1  
----------------
 あいう@\"@えお
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"+"', '+');
 pgs2snippet1 
--------------
 @+@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"+"', 'あいう+えお');
 pgs2snippet1  
---------------
 あいう@+@えお
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""+"', 'あいう"+えお');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"-"', '-');
 pgs2snippet1 
--------------
 @-@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"-"', 'あいう-えお');
 pgs2snippet1  
---------------
 あいう@-@えお
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""-"', 'あいう"-えお');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '" え"', 'あいう えお');
 pgs2snippet1  
---------------
 あいう@ え@お
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"" え"', 'あいう" えお');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, 'あ', '');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"あ"', '');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '+', '+"\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"', '+"\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '\\', '+"\\');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""', 'あ');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""', 'あ', 1);
ERROR:  function pgs2snippet1(integer, integer, integer, unknown, unknown, integer, unknown, unknown, integer) does not exist
LINE 1: SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '""', 'あ', 1);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"+"BB"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 @AA@"A+A\B+B"@BB@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"+""B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 @AA@"A+A\B+B@"B@B
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"+"+B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 @AA@"A+A\B@+B@"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"+"\\B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 @AA@"A+A@\B@+B"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A""+"BB"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 A@A"@A+A\B+B"@BB@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A""+""B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 A@A"@A+A\B+B@"B@B
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A""+"+B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 A@A"@A+A\B@+B@"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A""+"\\B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 A@A"@A+A@\B@+B"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A+"+"BB"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"@A+@A\B+B"@BB@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A+"+""B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"@A+@A\B+B@"B@B
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A+"+"+B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"@A+@A\B@+B@"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A+"+"\\B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"@A+@A@\B@+B"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A\\"+"BB"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"A+@A\@B+B"@BB@
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A\\"+""B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"A+@A\@B+B@"B@B
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A\\"+"+B"', 'AA"A+A\\B+B"BB');
   pgs2snippet1    
-------------------
 AA"A+@A\@B@+B@"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"A\\"+"\\B"', 'AA"A+A\\B+B"BB');
  pgs2snippet1   
-----------------
 AA"A+@A\@B+B"BB
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AB"CD"', 'AB"CD');
 pgs2snippet1 
--------------
 @AB"CD@
(1 row)

SET ludia_funcs.escape_snippet_keyword TO off;
SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AB"CD"', 'AB"CD');
 pgs2snippet1 
--------------
 @AB@"CD
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" + "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" + "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" + "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@@BBB@CCCDDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A"   "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A"   "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A"   "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@@BBB@CCCDDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" - "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@BBB@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" - "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@BBB@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA\\"A" - "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
    pgs2snippet1    
--------------------
 @AA"A@BBBCCCDDDEEE
(1 row)

SET ludia_funcs.escape_snippet_keyword TO on;
SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" + "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" + "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" + "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@@BBB@CCCDDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A"   "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A"   "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
      pgs2snippet1      
------------------------
 @AA"A@@BBB@@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A"   "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@@BBB@CCCDDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" - "BBB" + "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@BBB@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" - "BBB"   "CCC"', 'AA"ABBBCCCDDDEEE');
     pgs2snippet1     
----------------------
 @AA"A@BBB@CCC@DDDEEE
(1 row)

SELECT pgs2snippet1(1, 300, 1, '@', '@', 0, '"AA"A" - "BBB" - "CCC"', 'AA"ABBBCCCDDDEEE');
    pgs2snippet1    
--------------------
 @AA"A@BBBCCCDDDEEE
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,NULL,'あｲうｴおａbｃdｅかｷくｹこjｋlｍn');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'あｲうｴおａbｃdｅかｷくｹこjｋlｍn',NULL);
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'エおA','あｲうｴおａbｃdｅかｷくｹこjｋlｍn');
           pgs2snippet1            
-----------------------------------
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'えおA','あｲうｴおａbｃdｅかｷくｹこjｋlｍn');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'エおA',repeat(chr(13078),10) || chr(65018) || 'あｲうｴおａbｃdｅ' || chr(65018) || repeat(chr(13078),10) || 'かｷくｹこjｋlｍn' || chr(65018) || repeat(chr(13078),10));
                                           pgs2snippet1                                           
--------------------------------------------------------------------------------------------------
 ㌖㌖㌖㌖㌖㌖㌖㌖㌖㌖ﷺあｲう∇ｴおａ∇bｃdｅﷺ㌖㌖㌖㌖㌖㌖㌖㌖㌖㌖かｷくｹこjｋlｍnﷺ㌖㌖㌖㌖㌖㌖㌖㌖㌖㌖
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'""' || chr(13078) || chr(65018) || 'かキ""',chr(65018) || 'あｲうｴおａbｃdｅ' || chr(65018) || repeat(chr(13078),10) || 'かｷくｹこjｋlｍn' || chr(65018)||repeat(chr(13078),10));
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'"' || chr(13078) || chr(65018) || 'abｶ'|| '"','"' || repeat(chr(13078),10) || chr(65018) || 'あｲうｴおａbｃdｅ' || chr(65018) || repeat(chr(13078),10) || 'かｷくｹこjｋlｍn' || chr(65018) || repeat(chr(13078),10) || '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,3000,1,'∇','∇',0,'"' || chr(13078) || chr(65018) || '"','"' || repeat(chr(13078),10)|| repeat(chr(65018),10) || '"');
            pgs2snippet1            
------------------------------------
 "㌖㌖㌖㌖㌖㌖㌖㌖㌖∇㌖ﷺ∇ﷺﷺﷺﷺﷺﷺﷺﷺﷺ"
(1 row)

SELECT pgs2snippet1(1,3000,1,'∇','∇',0,'"' || repeat(chr(13078),10) || repeat(chr(65018),10) || '"','"' || chr(13078) || chr(65018)|| '"');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,repeat('x',300),'abcde');
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,repeat('+',300),'+');
WARNING:  too many expressions (236)
 pgs2snippet1 
--------------
 (null)
(1 row)

SELECT pgs2snippet1(1,300,1,'∇','∇',0,'エおA','あｲうｴおａbｃdｅかｷくｹこjｋlｍn') FROM generate_series(1,10);
           pgs2snippet1            
-----------------------------------
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
 あｲう∇ｴおａ∇bｃdｅかｷくｹこjｋlｍn
(10 rows)

